src/NuGet.Core/NuGet.Configuration/Settings/Items/PackageSourceNamespacesItem.cs

```diff
@@ -0,0 +1,233 @@
+        /// Each PackageSourceNamespaces item needs a key.
+        /// The key should correspond a package source key.
+        /// </summary>
+        public virtual string Key => Attributes[ConfigurationConstants.KeyAttribute];
+
+        protected override IReadOnlyCollection<string> RequiredAttributes { get; } = IReadOnlyCollectionUtility.Create(ConfigurationConstants.KeyAttribute);
+
+        protected void SetKey(string value)
+        {
+            if (string.IsNullOrEmpty(value))
+            {
+                throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Resources.PropertyCannotBeNullOrEmpty, nameof(Key)));
+            }
+
+            UpdateAttribute(ConfigurationConstants.KeyAttribute, value);
+        }
+
+        internal readonly IEnumerable<SettingBase> _parsedDescendants;
+
+        /// <summary>
+        /// Creates a package source namespace item with the given name, which equals the key and non-empty list of naemspace items.
+        /// </summary>
+        /// <param name="name">A non-empty name of the item which corresponds a package source name.</param>
+        /// <param name="namespaceItems">A non empty list of namespace items.</param>
+        /// <exception cref="ArgumentException">If <paramref name="name"/> is null or empty, or <paramref name="namespaceItems"/> is null or empty.</exception>
+        public PackageSourceNamespacesItem(string name, IEnumerable<NamespaceItem> namespaceItems)
+            : base()
+        {
+            if (string.IsNullOrEmpty(name))
+            {
+                throw new ArgumentException(Resources.Argument_Cannot_Be_Null_Or_Empty, nameof(name));
+            }
+
+            if (namespaceItems == null || !namespaceItems.Any())
+            {
+
+                throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Resources.Error_ItemNeedsAtLeastOneNamespace, name));
+            }
+
+            AddAttribute(ConfigurationConstants.KeyAttribute, name);
+
+            Namespaces = new List<NamespaceItem>();
+
+            foreach (var @namespace in namespaceItems)
+            {
+                Namespaces.Add(@namespace);
+            }
+        }
+
+        internal PackageSourceNamespacesItem(XElement element, SettingsFile origin)
+            : base(element, origin)
+        {
+            _parsedDescendants = element.Nodes().Where(n => n is XElement || n is XText text && !string.IsNullOrWhiteSpace(text.Value))
+                .Select(e => SettingFactory.Parse(e, origin));
+
+            var parsedNamespaceItems = _parsedDescendants.OfType<NamespaceItem>().ToList();
+
+            if (parsedNamespaceItems.Count == 0)
+            {
+                throw new NuGetConfigurationException(
+                    string.Format(CultureInfo.CurrentCulture, Resources.Error_ItemNeedsAtLeastOneNamespace, Key) +
+                    " " +
+                    string.Format(CultureInfo.CurrentCulture, Resources.Path, origin.ConfigFilePath)
```

Comment:
Avoid concatenating localized strings. It can lead to confusion in translated strings.

Add a new string resource if needed.

```suggestion
                    string.Format(CultureInfo.CurrentCulture, Resources.Error_AtLeastOneNamespaceWithPath, Key, origin.ConfigFilePath)
```

Reply:
> Avoid concatenating localized strings. It can lead to confusion in translated strings.

Can you elaborate? 

The `Path` string is pretty generic. 
The string with a full sentence is targeted and not split.

Reply:
> Can you elaborate?

See https://devdiv.visualstudio.com/DevDiv/_wiki/wikis/DevDiv.wiki/2440/Localizability-Do-and-Don't-Checklist

`Path` string looks pretty generic/straightforward in English. In other languages, 'path' word can have a lot more meanings.

Reply:
Pushed a change.